subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -0,0 +1,1445 @@
+	return 0;
+}
+
+#if defined(CONFIG_BT_MESH_CDB)
+SHELL_STATIC_SUBCMD_SET_CREATE(cdb_cmds,
+	/* Mesh Configuration Database Operations */
+	SHELL_CMD_ARG(create, NULL, "[NetKey]", cmd_cdb_create, 1, 1),
+	SHELL_CMD_ARG(clear, NULL, NULL, cmd_cdb_clear, 1, 0),
+	SHELL_CMD_ARG(show, NULL, NULL, cmd_cdb_show, 1, 0),
+	SHELL_CMD_ARG(node-add, NULL, "<UUID> <Addr> <Num-elem> "
+		      "<NetKeyIdx> [DevKey]", cmd_cdb_node_add, 5, 1),
+	SHELL_CMD_ARG(node-del, NULL, "<Addr>", cmd_cdb_node_del, 2, 0),
+	SHELL_CMD_ARG(subnet-add, NULL, "<NeyKeyIdx> [<NetKey>]",
+		      cmd_cdb_subnet_add, 2, 1),
+	SHELL_CMD_ARG(subnet-del, NULL, "<NetKeyIdx>", cmd_cdb_subnet_del,
+		      2, 0),
+	SHELL_CMD_ARG(app-key-add, NULL, "<NetKeyIdx> <AppKeyIdx> "
+		      "[<AppKey>]", cmd_cdb_app_key_add, 3, 1),
+	SHELL_CMD_ARG(app-key-del, NULL, "<AppKeyIdx>", cmd_cdb_app_key_del,
+		      2, 0),
+	SHELL_SUBCMD_SET_END);
+#endif
+
+SHELL_STATIC_SUBCMD_SET_CREATE(prov_cmds,
+	SHELL_CMD_ARG(uuid, NULL, "<UUID: 1-16 hex values>", cmd_uuid, 2, 0),
+	SHELL_CMD_ARG(input-num, NULL, "<Number>", cmd_input_num, 2, 0),
+	SHELL_CMD_ARG(input-str, NULL, "<String>", cmd_input_str, 2, 0),
+	SHELL_CMD_ARG(static-oob, NULL, "[Val: 1-16 hex values]", cmd_static_oob, 2, 1),
+	SHELL_CMD_ARG(local, NULL, "<NetKeyIndex> <Addr> [IVIndex]", cmd_provision_local,
+		      3, 1),
+	SHELL_CMD_ARG(beacon-listen, NULL, "<Val: off, on>", cmd_beacon_listen, 2, 0),
+
+/* Provisioning operations */
+#if defined(CONFIG_BT_MESH_PROV_DEVICE)
+#if defined(CONFIG_BT_MESH_PB_GATT)
+	SHELL_CMD_ARG(pb-gatt, NULL, "<Val: off, on>", cmd_pb_gatt, 2, 0),
+#endif
+#if defined(CONFIG_BT_MESH_PB_ADV)
+	SHELL_CMD_ARG(pb-adv, NULL, "<Val: off, on>", cmd_pb_adv, 2, 0),
+#endif
+#endif /* CONFIG_BT_MESH_PROV_DEVICE */
+
+#if defined(CONFIG_BT_MESH_PROVISIONER)
+	SHELL_CMD_ARG(remote-adv, NULL,
+		      "<UUID> <NetKeyIndex> <Addr> "
+		      "<AttentionDuration>",
+		      cmd_provision_adv, 5, 0),
+#endif
+
+#if defined(CONFIG_BT_MESH_PB_GATT_CLIENT)
+	SHELL_CMD_ARG(remote-gatt, NULL,
+		      "<UUID> <NetKeyIndex> <Addr> "
+		      "<AttentionDuration>",
+		      cmd_provision_gatt, 5, 0),
+#endif
+	SHELL_SUBCMD_SET_END);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(health_srv_cmds,
+	/* Health Server Model Operations */
+	SHELL_CMD_ARG(add-fault, NULL, "<Fault ID>", cmd_add_fault, 2, 0),
+	SHELL_CMD_ARG(del-fault, NULL, "[Fault ID]", cmd_del_fault, 1, 1),
+	SHELL_SUBCMD_SET_END);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(test_cmds,
+	/* Commands which access internal APIs, for testing only */
+	SHELL_CMD_ARG(net-send, NULL, "<Hex string>", cmd_net_send,
+		      2, 0),
+#if defined(CONFIG_BT_MESH_IV_UPDATE_TEST)
+	SHELL_CMD_ARG(iv-update, NULL, NULL, cmd_iv_update, 1, 0),
+	SHELL_CMD_ARG(iv-update-test, NULL, "<Val: off, on>", cmd_iv_update_test, 2, 0),
+#endif
+	SHELL_CMD_ARG(rpl-clear, NULL, NULL, cmd_rpl_clear, 1, 0),
+
+	SHELL_CMD(health-srv, &health_srv_cmds, "Low Power commands", bt_mesh_shl_mdl_cmds_help),
```

Comment:
I am inclined to not put them in models since these commands are dependent on the static shell implementation. I see that I have copy paste error on the description, so I will fix that. Do you not agree that these cmds are test-cmds? Any particular reason why you don't want them under test?

Reply:
That is what confuses me. Though the implementation in shell.c of these commands look like a test commands, they are useless if you have you own health server. As well as when you have more than health server.

Reply:
Right, I guess it will be tricky no matter where we put them. The only real solution that can be done for these commands is to clearly state in the documentation that you need to initialize the health server of the shell in the application to make this cmds work. an alternative would be a separate Kconfig option for this, so that we are certain that the user knowingly has initialized the health server and enabled the option. However, I'm not sure that this approach is the best.

Reply:
I'll create a separate task to discuss how to fix this.